crates/sui-types/src/messages.rs

```diff
@@ -403,7 +468,9 @@ impl TransactionKind {
                 let valid = self.single_transactions().all(|s| match s {
                     SingleTransactionKind::Call(_)
                     | SingleTransactionKind::TransferObject(_)
-                    | SingleTransactionKind::Pay(_) => true,
+                    | SingleTransactionKind::Pay(_)
+                    | SingleTransactionKind::PaySui(_)
+                    | SingleTransactionKind::PayAllSui(_) => true,
```

Comment:
I don't think these two can be in a batch, because of the specialty of gas object used.

Reply:
can you explain a bit why not? I thought as long as 2 txns do not have overlapping wrt objects, then they can be in a batch, so if paySui1 takes (obj1, obj2) and paySui2 takes (obj3, obj4), did I miss sth?

Reply:
First of all, they will be sharing the same first object (the gas object). So at least for PaySui, if it's included in a batch, you would have to make sure they all share the same first object. I am not quite sure how to chain the execution of multiple PaySui in the same tx, it may be doable, just complicated. And there is no strong motivation really, you should always be able to merge multiple PaySui from a batch transaction into one PaySui transaction.

For PayAllSui, because the gas object is fully transferred over, you certainly cannot have multiple PayAllSui in one batch, otherwise how do you determine which recipient to send the gas object?

I also suspect the current implementation doesn't pass the transaction input check, because the input objects contain the gas object, which is why the `TransferSui` transaction data doesn't include the gas object. You will be able to verify this part once you start e2e testing.